workers/unity/Packages/com.improbable.gdk.tools/GdkToolsConfiguration.cs

```diff
@@ -55,26 +54,32 @@ internal List<string> Validate()
                 errors.Add($"{GdkToolsConfigurationWindow.DescriptorOutputDirLabel} cannot be empty.");
             }
 
-            var emptySchemaSourceDir = false;
-            foreach (var schemaSourceDir in SchemaSourceDirs)
+            for (var i = 0; i < SchemaSourceDirs.Count; i++)
             {
+                var schemaSourceDir = SchemaSourceDirs[i];
+
                 if (!string.IsNullOrEmpty(schemaSourceDir))
                 {
-                    var fullSchemaSourceDirPath = Path.Combine(UnityProjectRoot, schemaSourceDir);
+                    string fullSchemaSourceDirPath;
+                    try
+                    {
+                        fullSchemaSourceDirPath = Path.Combine(UnityProjectRoot, schemaSourceDir);
+                    }
+                    catch (ArgumentException)
+                    {
+                        errors.Add($"Schema path [{i}] contains one or more invalid characters.");
+                        continue;
+                    }
+
                     if (!Directory.Exists(fullSchemaSourceDirPath))
```

Comment:
Move this into the try? Then you don't have to forward declare the `fullSchemaSourceDirPath` ?